server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -750,6 +779,16 @@ public void sendResponse(RestResponse response) {
             boolean success = false;
             try {
                 close();
+                methodHandlers.addRequestStats(contentLength);
+                methodHandlers.addResponseTime(rawRelativeTimeInMillis() - startTime);
+                if (response.isChunked() == false) {
+                    methodHandlers.addResponseStats(response.content().length());
+                } else {
+                    response = RestResponse.chunked(
+                        response.status(),
+                        new EncodeLengthTrackingChunkedRestResponseBody(response.chunkedContent(), methodHandlers::addResponseStats)
```

Comment:
Maybe just pass in `methodHandlers`, no need to allocate another method reference here.